### DISCUSSION TITLE
Can we have LTS version of images?

### DISCUSSION THREAD

**Discussion by lzqlzzq** [actions/virtual-environments#9593] [Ideas]
In last deployment, our project suffered from `glibc` version and `cmake` version: https://github.com/Natooz/MidiTok/issues/155.
Can we have LTS version of images for better compatibility and stability? Or specific version? Rolling release seems to be not a good practice for production environment.

**Comment by shamil-mubarakshin** [comment]
Hey @lzqlzzq. We are continuously striding towards a balance between providing images with latest versions and patches of software and keeping releases as stable as possible. We do understand concerns coming with this approach, but, unfortunately, there are no current plans on adding LTS versions of images.
When stability is paramount, [self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) can be considered as an option

**Comment by nebuk89** [comment]
Hey! We are working on a capability that will allow you to pin to certain versions of our images, this will come out with some wider options to customize what is on our images pre-job starting later this year :) watch this space <3

**Comment by lzqlzzq** [comment]
Thanks for your replies! That will do.

**Comment by lzqlzzq** [comment]
I think I should reopen this thread. Now we are suffered from another Visual Studio problem caused by the version(https://developercommunity.visualstudio.com/t/Internal-compiler-error-w-nanobind-v19/10662883?sort=newest).
I think that is really strange there is no version control for runner images. The Github itself IS essentially a website for version control.

**Comment by lzqlzzq** [comment]
Hey! Is there any progress? @nebuk89
